biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java

```diff
@@ -315,6 +320,31 @@ private void doRequireCapability(Annotation a) throws Exception {
 		Attrs attrs = getAttributes(a, "ns");
 		directivesAndVersion(attrs, "filter", "effective", "resolution");
 		
+		for (Entry<String,String> entry : attrs.entrySet()) {
+			String value = entry.getValue();
+
+			Matcher matcher = SIMPLE_PARAM_PATTERN.matcher(value);
+			while (matcher.find()) {
```

Comment:
I think you want to use a loop like:

    StringBuffer result = new StringBuffer();
     while (m.find()) {
        m.appendReplacement(result, attrs.get(m.group()));
    }
    entry.setValue(m.appendTail(result).toString());

It is more efficient.